modules/oe_bootstrap_theme_paragraphs/oe_bootstrap_theme_paragraphs.module

```diff
@@ -0,0 +1,591 @@
+  $variables['image'] = ImageValueObject::fromStyledImageItem($thumbnail, 'oe_bootstrap_theme_medium_no_crop');
+  // Overwrite the variant because is an image.
+  $variables['variant'] = 'image';
+  $cacheability->applyTo($variables);
+}
+
+/**
+ * Sets the embedded media.
+ *
+ * @param bool $is_image
+ *   Flag to determine if is an image or not.
+ * @param array $variables
+ *   The render array.
+ * @param Drupal\media\Entity\Media $media
+ *   Media object.
+ * @param Drupal\Core\Cache\CacheableMetadata $cacheability
+ *   CacheableMetadata object.
+ * @param Drupal\media\MediaSourceInterface $source
+ *   Media source.
+ */
+function oe_bootstrap_theme_set_embedded_media(bool $is_image, array &$variables, Media $media, CacheableMetadata $cacheability, MediaSourceInterface $source): void {
+  if ($is_image) {
+    return;
+  }
+
+  // Default video aspect ratio is set to 16x9.
+  $variables['ratio'] = '16x9';
+
+  // Load information about the media and the display.
+  $media_type = \Drupal::entityTypeManager()->getStorage('media_type')->load($media->bundle());
+  $cacheability->addCacheableDependency($media_type);
+  $source_field = $source->getSourceFieldDefinition($media_type);
+  $display = EntityViewDisplay::collectRenderDisplay($media, 'default');
+  $cacheability->addCacheableDependency($display);
+  $display_options = $display->getComponent($source_field->getName());
+
+  // If it is an OEmbed resource, render it and pass it as embeddable data
+  // only if it is of type video or html.
+  if ($source instanceof OEmbed) {
+    $oembed_type = $source->getMetadata($media, 'type');
+    if (in_array($oembed_type, ['video', 'html'])) {
+      $variables['embedded_media'] = $media->{$source_field->getName()}->view($display_options);
+      $cacheability->applyTo($variables);
+      return;
+    }
+  }
+
+  // If its an AvPortal video or an iframe video, render it.
+  $variables['embedded_media'] = $media->{$source_field->getName()}->view($display_options);
+  $cacheability->applyTo($variables);
+  // When dealing with iframe videos, also respect its given aspect ratio.
+  oe_bootstrap_theme_set_video_iframe_ratio($media, $variables, $cacheability);
+}
+
+/**
+ * Sets the video iframe ratio.
+ *
+ * @param Drupal\media\Entity\Media $media
+ *   Media object.
+ * @param array $variables
+ *   The render array.
+ * @param Drupal\Core\Cache\CacheableMetadata $cacheability
+ *   CacheableMetadata object.
+ */
+function oe_bootstrap_theme_set_video_iframe_ratio(Media $media, array &$variables, CacheableMetadata $cacheability): void {
+  if ($media->bundle() !== 'video_iframe') {
+    return;
+  }
+  $ratio = $media->get('oe_media_iframe_ratio')->value;
+  $variables['ratio'] = str_replace('_', 'x', $ratio);
+  $cacheability->applyTo($variables);
+}
+
+/**
+ * Implements hook_preprocess_paragraph().
+ */
+function oe_bootstrap_theme_preprocess_paragraph__oe_facts_figures(array &$variables): void {
```

Comment:
Move this with all  preprocess paragraphs function to be all together, let the auxiliary function to the bottom of the file